sql/src/main/java/io/crate/expression/scalar/systeminformation/CurrentSchemasFunction.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar.systeminformation;
+
+import com.google.common.collect.ImmutableList;
+import io.crate.data.Input;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.expression.symbol.Function;
+import io.crate.expression.symbol.format.FunctionFormatSpec;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.FunctionName;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.pgcatalog.PgCatalogSchemaInfo;
+import io.crate.types.DataTypes;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class CurrentSchemasFunction extends Scalar<String[], Object> implements FunctionFormatSpec {
+
+    private static final String NAME = "current_schemas";
+    private static final FunctionName FQN = new FunctionName(PgCatalogSchemaInfo.NAME, NAME);
+
+    public static final FunctionInfo INFO = new FunctionInfo(
+        new FunctionIdent(FQN, ImmutableList.of(DataTypes.BOOLEAN)),
+        DataTypes.STRING_ARRAY,
+        FunctionInfo.Type.SCALAR,
+        Collections.emptySet());
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(new CurrentSchemasFunction());
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return INFO;
+    }
+
+    @Override
+    public String[] evaluate(TransactionContext txnCtx, Input<Object>... args) {
+        boolean includeImplicitSchemas = true;
+        if (args.length > 0 && args[0] != null) {
+            includeImplicitSchemas = (Boolean) args[0].value();
```

Comment:
looks like this will result in an NPE if current_schemas(null) is used

Reply:
`&& args[0] != null` should protect it

Reply:
That only guards against a framework bug (inputs themselves should never be null). The check would have to be against `args[0].value()` - the boxing from `Boolean` to `boolean` causes the NPE

Reply:
right right stupid me, fixup pushed.